core/slice.go

```diff
@@ -739,59 +741,66 @@ func (sl *Slice) CollectNewlyConfirmedEtxs(block *types.WorkObject, location com
 			sl.hc.subRollupCache.Add(block.Hash(), subRollup)
 		}
 	}
-
 	// Filter for ETXs destined to this slice
-	newInboundEtxs := subRollup.FilterToSlice(location, nodeCtx)
-
+	newInboundEtxs := subRollup.FilterToSlice(blockLocation)
 	// Filter this list to exclude any ETX for which we are not the crossing
-	// context node. Such ETXs cannot be used by our subordinate for one of the
-	// following reasons:
-	// * if we are prime, but common dom was a region node, than the given ETX has
-	//   already been confirmed and passed down from the region node
-	// * if we are region, but the common dom is prime, then the destination is
-	//   not in one of our sub chains
-	//
-	// Note: here "common dom" refers to the highes context chain which exists in
-	// both the origin & destination. See the definition of the `CommonDom()`
-	// method for more explanation.
+	// context node. This check prevents cross-Region ETXs from going through Prime.
 	newlyConfirmedEtxs := newInboundEtxs.FilterConfirmationCtx(nodeCtx, nodeLocation)
```

Comment:
> This check prevents cross-Region ETXs from going through Prime.

Why would a cross-region ETX need to go through prime? It should only go through the region